src/prof_data.c

```diff
@@ -405,8 +406,10 @@ prof_lookup(tsd_t *tsd, prof_bt_t *bt) {
 
 prof_tctx_t *
 prof_tctx_create(tsd_t *tsd) {
-	prof_tdata_t *tdata = prof_tdata_get(tsd, false);
-	assert(tdata != NULL);
+	prof_tdata_t *tdata = prof_tdata_get(tsd, true);
+	if (tdata == NULL) {
```

Comment:
Should we do this instead? 
https://github.com/jemalloc/jemalloc/blob/4afd709d1f3ae7a727f144a96d8b834157d31e17/include/jemalloc/internal/prof_inlines_b.h#L91

Reply:
We also need some reentrancy level guard — https://github.com/jemalloc/jemalloc/blob/4afd709d1f3ae7a727f144a96d8b834157d31e17/include/jemalloc/internal/prof_inlines_b.h#L26

Reply:
I wonder what the possibilities are for `(uintptr_t)tdata <= (uintptr_t)PROF_TDATA_STATE_MAX` to be true. I cannot find any occurrences in the code base for `PROF_TDATA_STATE_REINCARNATED` or `PROF_TDATA_STATE_PURGATORY`.

On the other hand it seems to me that `prof_tdata_get()` either returns a valid `tdata` (i.e. a valid pointer to a real `tdata`) or `NULL`.

Reply:
If you are sure the PROF_TDATA_STATE_PURGATORY etc are not needed anymore, let’s clean up and make it consistent soon.

Reply:
It seems that the two "abnormal" `tdata` states got deleted in this commit: https://github.com/jemalloc/jemalloc/commit/5460aa6f6676c7f253bfcb75c028dfd38cae8aaf , but there didn't seem to be any explanations.